api/numaresourcesoperator/v1/helper/nodegroup/nodegroup.go

```diff
@@ -72,6 +75,13 @@ func FindTrees(mcps *mcov1.MachineConfigPoolList, nodeGroups []nropv1.NodeGroup)
 			return nil, fmt.Errorf("failed to find MachineConfigPool for the node group with the selector %q", nodeGroup.MachineConfigPoolSelector.String())
 		}
 
+		// generate name if not provided
+		trimmedName := strings.TrimSpace(nodeGroup.Name)
+		trimmedName = strings.Replace(trimmedName, " ", "", -1)
+		if trimmedName == "" {
+			nodeGroup.Name = guuid.New().String()
+		}
+
```

Comment:
let's do this in the normalize package. Once the code is moved should be OK conceptually. I'm not sure we would use a uuid here.